The Commit For Server Security

Goblom opened the pull request:
> Added saftey for servers that do not want maps downloaded 
> @Vareide @Clankstar @jollyolbrits @OvercastNetwork @ItsHarry @lazertester

Some servers have maps that they do not want downloaded (Example: TheHives SG6).
This commit allows server admins to be able to secure their maps and make sure they don't get released to the public unless they want them to. 

**The security is disabled by default and it requires the server admin to add one of the two codes below to their server.properties MOTD.**
- \u00A7-
- \u00a7-

**TODO**
- [x] Use MOTD to allow server admins to keep worlds private.
- [ ] Instead of sending a message to the users chat, just disable the button.
- [ ] Read from in-game MOTD 
- [ ] Maybe send message through plugin channels to server admins telling them that user has World Downloader Mod if a certain plugin is installed ?????? Not entirely sure.

Goblom committed:
Added safety for servers to disabled downloading

Added saftey for servers that do not want maps downloaded @Vareide
@Clankstar @jollyolbrits @OvercastNetwork @ItsHary @lazertester

lwatsondev commented:
You sir, are a god.

Lunatrius commented on src/WDL.java:
```diff
@@ -96,14 +96,28 @@
+    public static void preStart( )
+    {
+    	String isPrivateServer = mc.getServerData().serverMOTD;
+    	if (!isPrivateServer.contains("�-") || !isPrivateServer.contains("\u00A7-") || !isPrivateServer.contains("\u00a7-"))
```
You do realize that all 3 checks mean the same thing ("§" == "\u00A7-" == "\u00a7-")? Only thing that's really different is the notation. I'd use the hex notation though, simply because the file doesn't have to be UTF-8 to get the correct symbol (because of an invalid encoding it's actually "�-").

drumming102 commented:
We love you for this

Goblom committed:
Did some testing with the information @Lunatrius said

He is completely right. The § symbol is automatically converted to hex
upon server start.

Goblom commented:
@james-gibson I have encountered both, so i added both, just in case.

Lunatrius commented:
It doesn't really matter. Hexadecimal numbers a-f are case insensitive. 0xff is the same as 0xFF. Everything \uxxxx does is to insert the character with the code xxxx at that position.

dslake commented:
Why would anyone want to install this version of the mod on their client? Putting this code in master is just asking someone to fork my project, remove the check and distribute their own binaries.
